src/components/Header/Header.js

```diff
@@ -27,7 +29,31 @@ const HeaderWrapper = styled.header`
 `;
 
 class Header extends Component {
+  constructor(props) {
+    super(props);
+    this.handleCLick = this.handleCLick.bind(this);
+  }
+
+  handleCLick() {
+    if (localStorage.getItem("uid") !== null) {
```

Comment:
Puedes aprovechar los falsy y truthy de JS y escribir 

```suggestion
    if (localStorage.getItem("uid")) {
```

Si recibes algo  sera true y si no false.